File: interpreter/src/pregen/scala/org/enso/interpreter/nodes/expression/operator/ModOperatorNode.scala
package org.enso.interpreter.nodes.expression.operator

import com.oracle.truffle.api.dsl.Specialization
import com.oracle.truffle.api.nodes.NodeInfo
import org.enso.interpreter.nodes.expression.BinaryOperatorNode

@NodeInfo(shortName = "%")
abstract class ModOperatorNode extends BinaryOperatorNode {

  @Specialization
  protected def mod(left: Long, right: Long): Long =
    left % right
}

File: interpreter/src/main/java/org/enso/interpreter/nodes/expression/operator/ModOperatorNode.java
package org.enso.interpreter.nodes.expression.operator;

import com.oracle.truffle.api.dsl.Specialization;
import com.oracle.truffle.api.nodes.NodeInfo;
import org.enso.interpreter.nodes.expression.BinaryOperatorNode;

@NodeInfo(shortName = "%")
public abstract class ModOperatorNode extends BinaryOperatorNode {

    @Specialization
    protected long mod(long left, long right) {
        return left % right;
    }
}
